app/models/manageiq/providers/google/cloud_manager/event_catcher/runner.rb

```diff
@@ -0,0 +1,74 @@
+class ManageIQ::Providers::Google::CloudManager::EventCatcher::Runner <
+  ManageIQ::Providers::BaseManager::EventCatcher::Runner
+
+  # Request the event monitor to stop running.
+  def stop_event_monitor
+    stop_running
+  end
+
+  # Start monitoring for events. This method blocks forever until
+  # #stop_event_monitor is called.
+  def monitor_events
+    _log.info "#{log_prefix} Monitoring for events"
+
+    start_running
+    event_monitor_running
+    while running?
+      heartbeat
+      events = event_producer.poll
+      heartbeat
+
+      _log.info "#{log_prefix} Received #{events.size} events" unless events.empty?
+      events.each do |event|
+        _log.debug "#{log_prefix} Event: #{event}"
+        @queue.enq event if event_requires_refresh?(event)
```

Comment:
@selmanj Typically what we'll do here is use event blacklisting to ignore specific event types (see the `blacklisted_events` table).

Then, all other events just get queued here.  And, even if the event doesn't result in a specific action to be taken (like refreshing), we would still capture the event in order to show the event on timeline reports.

If there's no way to black list the specific events (way too many to list, or you simply don't know the list of events), then we might be able to get away with this whitelisting via matching.

@Fryguy, thoughts on this?